src/views/W3NCreateSign/W3NCreateSign.tsx

```diff
@@ -70,17 +70,9 @@ export function W3NCreateSign({
     address,
   });
 
-  const fullDidDetails = useSwrDataOrThrow(
-    did,
-    getFullDidDetails,
-    'getFullDidDetails',
-  );
-  const fee = useSwrDataOrThrow(fullDidDetails, getFee, 'W3NCreateSign.getFee');
-  const deposit = useSwrDataOrThrow(
-    did,
-    getDepositWeb3Name,
-    'W3NCreateSign.getDeposit',
-  )?.amount;
+  const fullDidDetails = useFullDidDetails(did);
+  const fee = useAsyncValue(getFee, [fullDidDetails]);
```

Comment:
Since we have `useDeposit` hooks, what about a `useFee` hook as well?

Reply:
This particular function is used in this file only, we could make the test more readable this way, but the code would become less readable. I think the code takes the precedence here :)